example/storybook/stories/Notifications/NotificationsScreen.stories.tsx

```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react-native';
+import { NotificationsScreen } from '.';
+import { PushNotificationsProvider } from 'src/hooks/usePushNotifications';
+
+storiesOf('NotificationsScreen', module).add('demo', () => (
+  <PushNotificationsProvider>
```

Comment:
This can be improved by moving the provider up the component tree to live with the rest in RootProviders. If we want to drop the PushNotificationsProvider into RootProviders, we are running into an issue where the context is undefined. 

The only fix that I could come up with to resolve the context issue was to move the initialization logic back into the component which defeated the purpose of the provider in the first place. So I'll need to lean into yall expertise about the best way to resolve this issue.

Reply:
I think this is fine, if we are aiming to have Notifications "opt-in" rather than enabled by default. If I were a consumer of this lib, I would look at this story and say, "okay, I need to wrap my component in a provider to get it to work".

The reason you are getting context being undefined, is because you set the default to `undefined` [here](https://github.com/lifeomic/react-native-sdk/pull/238/files#diff-21335a07ddbab7e65d193bfa347f491ddcab07229361b14275778f3e0d92d759R23). Also why you have the override [here](https://github.com/lifeomic/react-native-sdk/pull/238/files#diff-1f54bc7376235dcc4820ec4b651589e27ef330fe6f71a5dcf182a3682cdad53cR60).

The simple fix to this would be to initialize the context, something like
```typescript
export const PushNotificationsContext =
  createContext<PushNotificationsStateType>({
    events: [],
    setEvents: () => {},
    httpClient: defaultAxiosInstance,
    account: undefined,
  });
```

But now looking closer at this type, I don't think you should be including the `httpClient` or `account` in the context, since these are readily available as their own hooks. The context should just be about notifications, then the implementation can include the hook(s).